pkg/sparrow/run.go

```diff
@@ -141,6 +143,49 @@ func (s *Sparrow) ReconcileChecks(ctx context.Context) {
 	}
 }
 
+// updateCheckTargets updates the targets of a check with the
+// global targets. The targets are merged per default, if found in the
+// passed config.
+func (s *Sparrow) updateCheckTargets(cfg any) any {
+	if cfg == nil {
+		return nil
+	}
+
+	checkCfg, ok := cfg.(map[string]any)
+	if !ok {
+		return checkCfg
+	}
+	if _, ok = checkCfg["targets"]; !ok {
+		return checkCfg
+	}
+
+	actuali, ok := checkCfg["targets"].([]any)
+	if !ok {
+		return checkCfg
+	}
+
+	var actual []string
+	for _, v := range actuali {
+		actual = append(actual, v.(string))
+	}
+	var urls []string
+	gt := s.targets.GetTargets()
+
+	// filter out globalTargets that are already in the config and self
+	for _, t := range gt {
+		if slices.Contains(actual, t.Url) {
+			continue
+		}
+		if t.Url == fmt.Sprintf("https://%s", s.cfg.SparrowName) {
```

Comment:
what if sparrow runs on http?

Reply:
Then it's a problem for another time :) 

We don't have this requirement ATM. If such a requirement comes, it can be easily tackled by letting the sparrow know about the protocol, or checking both encrypted & unencrypted HTTP traffic.